src/rpc/response.zig

```diff
@@ -1,27 +1,26 @@
 const std = @import("std");
 
-const Allocator = std.mem.Allocator;
-
 pub const ParseError = std.json.ParseError(std.json.Scanner) || error{MissingResult};
 
 /// Wraps a parsed response from the RPC server with an arena that owns all
 /// contained pointers.
-pub fn Response(comptime Method: type) type {
+pub fn Response(comptime T: type) type {
     return struct {
         arena: *std.heap.ArenaAllocator,
         id: u64,
         jsonrpc: []const u8,
         payload: Payload,
+        const Self = @This();
 
         pub const Payload = union(enum) {
-            result: Method.Response,
+            result: T,
```

Comment:
Is this needed for the server?

Reply:
It isn't needed for the server per se, but I made this change because the original way it was structured confused me for quite a bit. Having `Response` simply be generic over the response type directly, rather than "a type that exposes a type decl called 'Response', which will be the only thing that is actually used" seems like a natural decoupling.

Reply:
Interesting. I find `Response(GetAccountInfo)` to be very intuitive. It's a function that returns the response type for the GetAccountInfo method. Now it's used like `Response(GetAccountInfo.Response)`, and returns the response type for a method's response type? It feels more confusing/redundant/verbose to me now.

Anyway, obviously I'm going to have this opinion, since I'm the one who deliberately made the original design choice. This change feels a bit divisive, but relatively inconsequential compared to all the improvements you've made, so I'll be happy to approve the PR regardless.